### PR TITLE
Updated Sun providers according to the algorithm

### PR DESCRIPTION
**Pull request by una-tapa**

### CODE REVIEW

**Comment by una-tapa** [comment]
Fixes #2219

**Comment by una-tapa** [comment]
#build

**Comment by una-tapa** [comment]
The build completed with some errors in  cdi,concurrent.persistent,jaxrs client, most of them looks timing issue.  Not LTPA related error found.   I am going to ask reviewers to help double check.

**Comment by una-tapa** [comment]
Had a chance to run with both OracleJVM and OpenJDK locally.  Unittest issues were no longer seen.

File: dev/com.ibm.ws.security.token.ltpa/src/com/ibm/ws/security/token/ltpa/internal/LTPAToken2.java [hunk 1]
```diff
@@ -62,11 +62,29 @@
...
+    private static boolean ibmJVM = false;
+    
+
     static {
         MessageDigest m1 = null, m2 = null;
         try {
-	    m1 = MessageDigest.getInstance(MESSAGE_DIGEST_ALGORITHM, LTPAKeyUtil.defaultJCEProvider());
-	    m2 = MessageDigest.getInstance(MESSAGE_DIGEST_ALGORITHM, LTPAKeyUtil.defaultJCEProvider());
+	    if (javaVendor == null) {
+		javaVendor = System.getProperty("java.vendor");
```
Comment (aguibert) [inline dev/com.ibm.ws.security.token.ltpa/src/com/ibm/ws/security/token/ltpa/internal/LTPAToken2.java]: Consider using the `JavaInfo` API (from kernel.service) for checking the java vendor.

Comment (una-tapa) [reply dev/com.ibm.ws.security.token.ltpa/src/com/ibm/ws/security/token/ltpa/internal/LTPAToken2.java]: Will update the code and run the build again.

**Comment by una-tapa** [comment]
Pls do not merge just yet.  Added Andy's suggestion on the token package not util.  Will run the build once more.

**Comment by una-tapa** [comment]
#build

**Comment by una-tapa** [comment]
Just need to confirm the change in the build...  I will ask for merge when I see it ok.

**Comment by ewhoch** [comment]
all part of issue #2068

**Pull request merged**